data/eos-updater-autoinstall.schema.json

```diff
@@ -0,0 +1,138 @@
+            ]
+        },
+        "serial": {
+            "description": "A unique serial number for this event, unique to the domain named after the autoinstall file. Serial numbers must be assigned monotonically increasing values.",
+            "type": "integer"
+        },
+        "ref-kind": {
+            "description": "The flatpak ref kind",
+            "type": "string",
+            "enum": ["app", "runtime"]
+        },
+        "app": {
+            "description": "The app-id to be installed",
+            "type": "string"
+        },
+        "collection-id": {
+            "description": "The collection-id to install the app from",
+            "type": "string"
+        },
+        "filters": {
+            "type": "object",
+            "title": "Detail for Flatpak Removal",
+            "description": "A description of a flatpak to be automatically removed",
+            "properties": {
+                "action": {
+                    "const": "uninstall",
+                    "description": "The machine-readable type of event",
+                    "type": "string"
+                },
+                "serial": {
+                    "$ref": "#/definitions/serial"
+                },
+                "filters": {
+                    "$ref": "#/definitions/filters"
+                },
+                "ref-kind": {
+                    "$ref": "#/definitions/ref-kind"
+                },
+                "app": {
+                    "$ref": "#/definitions/app"
+                }
+            }
+        }
+    },
+    "required": ["action", "serial", "detail"],
+    "definitions": {
+        "filters": {
+            "description": "Tests to apply to see whether this action should be performed or skipped. If it is skipped, the test to check whether it should be performed or skipped will never run again.",
+            "type": "object",
+            "oneOf": [
+                { "$ref": "#/definitions/filters" }
+            ]
+        },
+        "serial": {
+            "description": "A unique serial number for this event, unique to the domain named after the autoinstall file. Serial numbers must be assigned monotonically increasing values.",
+            "type": "integer"
+        },
+        "ref-kind": {
+            "description": "The flatpak ref kind",
+            "type": "string",
+            "enum": ["app", "runtime"]
+        },
+        "app": {
+            "description": "The app-id to be installed",
+            "type": "string"
+        },
+        "filters": {
+            "type": "object",
+            "title": "Filters to apply to action",
+            "description": "Tests to apply to see if action should be applied or skipped",
+            "properties": {
+                "~architectures": {
+                    "description": "Which architectures forbid performing this action on",
+                    "type": "array",
+                    "minItems": 1,
```

Comment:
`~architectures` can legitimately be empty (it would be a weird autoinstall entry, but it’s valid)